src/main.py

```diff
@@ -34,8 +34,11 @@
     )
     # create the empty json file for images
     build_context = os.getenv("BUILD_CONTEXT")
+    ei_dedicated = os.getenv("EIA_DEDICATED") == "True"
     utils.write_to_json_file(constants.TEST_TYPE_IMAGES_PATH, {})
-    if args.framework not in build_config.DISABLE_FRAMEWORK_TESTS or build_context != "PR":
+    if (not ei_dedicated and not build_config.ENABLE_EI_MODE and args.framework not in build_config.DISABLE_FRAMEWORK_TESTS) or \
```

Comment:
can we factor out this if logic into a function? it's a bit tough to read this condition when spelled out like this

Reply:
I might split this 3 "OR" condition into 3 temp variable with meaningful name